File: Programming with C#/1. C# Fundamentals I/4. Console Input-Output/06. Quadratic Equation/QuadraticEquation.cs
/*Problem 6. Quadratic Equation
Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).*/

using System;

class QuadraticEquation
{
    static void FormatingPrint(double x1, double x2)
    {
        bool checkX1 = Convert.ToString(x1).IndexOf(".") > 0;
        bool checkX2 = Convert.ToString(x2).IndexOf(".") > 0;
        string index1 = "1";
        string index2 = "2";
        if (x1 == x2)
        {
            Console.WriteLine(checkX1 ? "x{1} = x{2} = {0:0.0}" : "x{1} = x{2} = {0}", x1, index1, index2);
        }
        else
        {
            Console.Write(checkX1 ? "x{1} = {0:0.0}; " : "x{1} = {0}; ", x1, index1);
            Console.WriteLine(checkX1 ? "x{1} = {0:0.0}" : "x{1} = {0}", x2, index2);
        }

    }
    static void Main()
    {
        Console.Title = "Quadratic Equation";

        Console.WriteLine("Enter coefficients a, b, and c of a quadratic equation.");
        Console.WriteLine(new string('-', 60));

        Console.Write("Enter a --> ");
        double a;
        while (!double.TryParse(Console.ReadLine(), out a))
        {
            Console.Write("Invalid parameter! Enter a --> ");
        }

        Console.Write("Enter b --> ");
        double b;
        while (!double.TryParse(Console.ReadLine(), out b))
        {
            Console.Write("Invalid parameter! Enter b --> ");
        }

        Console.Write("Enter c --> ");
        double c;
        while (!double.TryParse(Console.ReadLine(), out c))
        {
            Console.Write("Invalid parameter! Enter c --> ");
        }

        Console.WriteLine(new string('-', 60));

        double determenat;
        double x1;
        double x2;

        if ((b * b) - (4 * a * c) < 0)
        {
            Console.WriteLine("no real roots");

        }
        else
        {
            determenat = Math.Sqrt((b * b) - (4 * a * c));
            x1 = (-b - determenat) / (2 * a);
            x2 = (-b + determenat) / (2 * a);
            FormatingPrint(x1, x2);
        }

        Console.WriteLine(new string('-', 60));
    }
}

File: Web Design and UI/3. JS-Fundamentals/HW/03. Conditional statements/scripts/quadraticEquation.js
/**
 Problem 6. Quadratic equation

 Write a script that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).
 Calculates and prints its real roots.
 Note: Quadratic equations may have 0, 1 or 2 real roots.

 Examples:

 a    b    c    roots
 2    5    -3    x1=-3; x2=0.5
 -1    3    0    x1=3; x2=0
 -0.5    4    -8    x1=x2=4
 5    2    8    no real roots
 */

var i,
    len,
    arrayA = [2, -1, -0.5, 5],
    arrayB = [5, 3, 4, 2],
    arrayC = [-3, 0, -8, 8];

for ( i = 0, len=arrayA.length; i < len; i+=1) {
    quadraticEquations(arrayA[i], arrayB[i], arrayC[i]);
}

function quadraticEquations(a, b, c) {
    var d,
        x,
        x1,
        x2;

    d = b * b - 4 * a * c;

    if (d < 0) {
        console.log('no real roots');
    } else if (d === 0) {
        x = -b / (2 * a);
        console.log('x1 = x2= ' + x);
    } else {
        x1 = (-b - Math.sqrt((b * b) - (4 * a * c))) / (2 * a);
        x2 = (-b + Math.sqrt((b * b) - (4 * a * c))) / (2 * a);
        console.log('x1 = ' + x1 + '; ' + 'x2 = ' + x2);
    }
}
